quic/s2n-quic-platform/src/io/xdp/builder.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::mem::size_of;
+use s2n_quic_core::{
+    inet::{ethernet, ipv4, udp},
+    path::{MaxMtu, MaxMtuError},
+};
+use tokio::runtime::Handle;
+
+/// Calculate how much a packet will need for fixed-size headers
+const MIN_FRAME_OVERHEAD: u16 =
+    (size_of::<ethernet::Header>() + size_of::<ipv4::Header>() + size_of::<udp::Header>()) as _;
+
+#[derive(Debug)]
+#[must_use = "Builders do nothing without calling `build`"]
+pub struct Builder<Rx = (), Tx = ()> {
+    rx: Rx,
+    tx: Tx,
+    max_mtu: MaxMtu,
+    handle: Option<Handle>,
+}
+
+impl Default for Builder<(), ()> {
+    fn default() -> Self {
+        Self {
+            rx: (),
+            tx: (),
+            max_mtu: MaxMtu::try_from(4096 - MIN_FRAME_OVERHEAD).unwrap(),
+            handle: None,
+        }
+    }
+}
+
+impl<Rx, Tx> Builder<Rx, Tx> {
+    /// Sets the tokio runtime handle for the provider
+    pub fn with_handle(mut self, handle: Handle) -> Self {
+        self.handle = Some(handle);
+        self
+    }
+
+    /// Sets the UMEM frame size for the provider
+    pub fn with_frame_size(mut self, frame_size: u16) -> Result<Self, MaxMtuError> {
```

Comment:
Is there any additional validation you could perform here?

Reply:
At the moment, probably not? At this point the application would have already constructed the UMUM. This is just for informing the endpoint of the MTU for the controller.